django_mongodb_backend/base.py

```diff
@@ -176,7 +177,12 @@ def get_connection_params(self):
 
     @async_unsafe
     def get_new_connection(self, conn_params):
-        return MongoClient(**conn_params, driver=self._driver_info())
+        if self.alias not in self._connection_pools:
+            conn = MongoClient(**conn_params, driver=self._driver_info())
+            # setdefault() ensures that multiple threads don't set this in
+            # parallel.
+            self._connection_pools.setdefault(self.alias, conn)
+        return self._connection_pools[self.alias]
```

Comment:
Is it ever possible for this cache to become out of sync? Does the user ever have raw access to the MongoClient (where they could call `client.close()`)?

Reply:
Yes, the `MonogClient` is available at `django.db.connection.connection`, but I don't think it's our job to consider situations where the user does something unexpected like calling `close()`.